rest-server/src/components/workouts/workoutQueries.js

```diff
@@ -15,5 +20,18 @@ export const workoutQuery = async (payload, url) => {
     return await globalQueryHelper(payload, fetchExercisesByWorkout, 'fetchExercisesByWorkout', ['workout_id']);
   } else if (url.slice(0,5) === '/user') {
     return await globalQueryHelper(payload, fetchWorkoutsByUser, 'fetchWorkoutsByUser', ['user_id']);
+  } else if (url.slice(0, 7) === '/delete') {
+    // query for exercises by workout id
+    const exercises = await globalQueryHelper(payload, fetchExerciseIdsByWorkout, 'fetchExerciseIdsByWorkout', ['workout_id']);
+    await globalQueryHelper(payload, deleteFromExerciseWorkout, 'deleteFromExerciseWorkout', ['workout_id']);
+    await globalQueryHelper(payload, deleteFromUsersWorkouts, 'deleteFromUsersWorkouts', ['workout_id']);
+    // iterate over exercises and delete each from exercises
+    let i = 0;
+    while (i < exercises.rows.length) {
```

Comment:
can you put this in a try/catch?

It's dangerous to have possibly failing requests on the rest-server.

The while loop is probably a larger one.